### DISCUSSION TITLE
Understanding log: lines repeated n times

### DISCUSSION THREAD

**Discussion by rrodrigueznt** [abraunegg/onedrive#2294] [Q&A] [answered]
Hi! Working with ```onedrive v2.4.23``` on a ```CentOS Stream 8``` server. Happy to see how OneDrive works great and struggling to understand the details.

A OneDrive instance working with the following configuration:

```
(base) [rjr@e220i059h057w ~]$ cat .config/onedrive_usc-bsf/config | grep -v '^#'
sync_dir = "~/OneDrive_USC-BSF"
log_dir = "/var/log/onedrive_usc-bsf/"
download_only = "true"
cleanup_local_files = "true"
enable_logging = "true"
sync_business_shared_folders = "true"
```

I get log records for each excluded folder by ```sync_list``` config repeated several times for different folders. For instance, for the number of appearances, some lines at the top and the bottom:

```
(base) [rjr@e220i059h057w ~]$ cat /var/log/onedrive_usc-bsf/rjr.onedrive.log | grep 'excluded' | sed 's/.*config//' | sort | uniq -c | sort -h -r
    216 : ./Documents
    200 : ./Pictures
    110 : ./0000_IGFAE_Prototypes
    102 : ./1000_RJR
     88 : ./0212_IGFAE_Software
     74 : ./Microsoft Teams Chat Files
     72 : ./0020_IGFAE_Procedures
     60 : ./1400_RJR_OpenAir
...
      2 : ./0011_IGFAE_IESIDE
      2 : ./0010_IGFAE_KeyDocuments
      2 : ./0006_IGFAE_Cigus
      2 : ./0005_IGFAE_Funding
      2 : ./0001_IGFAE_HRS4RatUSC
```

Please, why? Thanks in advance!

**Comment by abraunegg** [comment, marked as answer]
@rrodrigueznt 
> Please, why?

Because you are excluding that many items with your configuration. Please remember that you are doing 'client side filtering' and OneDrive online has no idea that you are doing client side filtering - this sends you all items / objects online - which you need to filter and are filtering with your respective configuration. This is why the application is informing you of each item it is skipping.

To reduce application logging of `Skipping item ...` reduce application verbosity by removing `--verbose` from your application configuration.

**Reply by rrodrigueznt** [reply]
> @abraunegg 
 
> To reduce application logging of `Skipping item ...` reduce application verbosity by removing `--verbose` from your application configuration.

No, no! I love verbosity as it provides a lot of information that helps me understand what is going on! The point is that if I have only one ```./Documents``` or one ```./Pictures``` folder, why does the log list 216 and 200 times these folders as excluded?

Thanks!

**Reply by abraunegg** [reply]
> why does the log list 216 and 200 times these folders as excluded

Because you have X number of items (files and folders) in that folder you are skipping

**Reply by rrodrigueznt** [reply]
@abraunegg 

The number of items is much higher here. For instance, ```./Documents``` contains 18154 files and  ```./Pictures``` contents 18239. The number of directories fits neither. Never mind, It won't be an issue! Thanks for your time! I'll keep trying to find what that number means and keep this thread posted if I find an answer or face any problems. Thanks!

**Reply by rrodrigueznt** [reply]
@abraunegg

> The number of items is much higher here. For instance, `./Documents` contains 18154 files and `./Pictures` contents 18239. The number of directories fits neither. Never mind, It won't be an issue! Thanks for your time! I'll keep trying to find what that number means and keep this thread posted if I find an answer or face any problems. Thanks!

I will add here a link to the response to this doubt in other of your posts. Thanks!

https://github.com/abraunegg/onedrive/discussions/2295#discussioncomment-4743046

**Comment by abraunegg** [comment]
@rrodrigueznt 
That question and commen is 100% not related to your question.

If you really want to know *why* the client is telling you why it is skipping something because of your explicit configuration to sync only certain items - please run the client in debug mode `--verbose --verbose` so that you can understand what the client is doing in your scenario.

I have already explained to you that your log output is due to ths items you are excluding. 

I dont know how else to give you a different answer to your question because I have already answered it.

**Reply by rrodrigueznt** [reply]
@abraunegg 

> I dont know how else to give you a different answer to your question because I have already answered it.

Yes! You already answered my question! And the reference to the 200-240 bundle lots helped me to understand that I must not read every line in the log independently! For instance, each line telling me that ```./Documents``` is excluded refers to a 200-240 bundle lot contained in this folder, not the whole folder. I perfectly understand now what I see here.

Thank you very much for your help and your great job!
